src/expr/node_manager.h

```diff
@@ -458,6 +458,17 @@ class NodeManager
   /** Create a node with an arbitrary number of children. */
   template <bool ref_count>
   Node mkNode(Kind kind, const std::vector<NodeTemplate<ref_count> >& children);
+
+  /** Create a node using an initializer list.
+   *
+   * This function serves two purposes:
+   * - We can avoid creating a temporary vector in some cases, e.g. when we
+   *   want to create a node with zero children
+   * - It makes sure that calls to `mkNode` that use braced-init-lists work as
+   *   expected, e.g., mkNode(REGEXP_EMPTY, {})
```

Comment:
If the motivation is a node with zero children, why don't we have `mkNode(Kind)`?
I'd rather see this new overload as an alternative to `mkNode()` with 1,2,3,4,5 children...

Reply:
I agree.

Reply:
I see the point regarding `{}` and I am happy to add another overload for it. However, the idea was that this function adds a way of creating a node with an arbitrary number of children without the need of creating a temporary vector (C-style variadic functions are [not recommended](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-ellipses)). I have considered using C++17 [fold expressions](https://en.cppreference.com/w/cpp/language/fold) instead of defining all these different variants but it adds about 3 minutes of CPU time to recompiling cvc5.